src/packages/core/components/alias-input/alias-input.element.ts

```diff
@@ -0,0 +1,112 @@
+import { UmbChangeEvent } from '@umbraco-cms/backoffice/event';
+import { css, html, customElement, property, state } from '@umbraco-cms/backoffice/external/lit';
+import { FormControlMixin, UUIInputEvent } from '@umbraco-cms/backoffice/external/uui';
+import { UmbLitElement } from '@umbraco-cms/internal/lit-element';
+
+@customElement('umb-alias-input')
+export class UmbAliasInput2Element extends FormControlMixin(UmbLitElement) {
+	protected getFormElement() {
+		return undefined;
+	}
+
+	@property()
+	public set value(val: string | null) {
+		if (val === null) return;
+		this.locked = true;
+		super.value = val;
+		this.pristine = false;
+	}
+	public get value(): string {
+		return super.value as string;
+	}
+
+	private _listenToElement: HTMLElement | null = null;
+
+	public set listenToElement(el: HTMLElement | null) {
+		if (this._listenToElement) {
+			this._listenToElement.removeEventListener('input', this.#parentChange);
+		}
+		this._listenToElement = el;
+		this._listenToElement?.addEventListener('input', this.#parentChange);
+	}
+	public get listenToElement(): HTMLElement | null {
+		return this._listenToElement;
+	}
+
+	@state()
+	locked = true;
+
+	#parentIsInput: boolean;
+
+	constructor() {
+		super();
+		this.listenToElement = (this.getHostElement() as Element).parentElement;
+		this.#parentIsInput = this.listenToElement?.nodeName === 'UUI-INPUT';
+	}
+
+	disconnectedCallback(): void {
+		if (this.parentElement) {
+			this.parentElement.removeEventListener('input', this.#parentChange);
+		}
+	}
+
+	#onToggleAliasLock() {
+		this.locked = !this.locked;
+	}
+
+	#onChange(e: UUIInputEvent) {
+		super.value = e.target.value as string;
```

Comment:
The current class is in itself a form control with a setter for the value, but you are setting the value only on the super element thereby bypassing the checks you have made in the "set value" function. Is that intended?